_15thnight/__init__.py

```diff
@@ -4,44 +4,74 @@
 from flask.ext.login import LoginManager, current_user, logout_user
 from werkzeug.datastructures import MultiDict
 
-from _15thnight import database, queue
-from _15thnight.api import (
-    account_api, alert_api, category_api, need_api, response_api, service_api,
-    user_api
-)
 from _15thnight.email import mailer
 from _15thnight.forms import csrf_protect
 from _15thnight.models import User
 from _15thnight.util import ExtensibleJSONEncoder
 
 
-app = Flask(__name__)
-
-try:
-    app.config.from_object('config')
-except:
-    app.config.from_object('configdist')
+login_manager = LoginManager()
 
-app.secret_key = app.config['SECRET_KEY']
-app.json_encoder = ExtensibleJSONEncoder
 
-app.register_blueprint(alert_api, url_prefix='/api/v1')
-app.register_blueprint(account_api, url_prefix='/api/v1')
-app.register_blueprint(category_api, url_prefix='/api/v1')
-app.register_blueprint(need_api, url_prefix='/api/v1')
-app.register_blueprint(response_api, url_prefix='/api/v1')
-app.register_blueprint(service_api, url_prefix='/api/v1')
-app.register_blueprint(user_api, url_prefix='/api/v1')
+def create_app(cfg=dict()):
+    """
+    Setup an app with blueprints, extensions, and other necessary additions.
 
-queue.init_app(app)
+    cfg_dict: dict
+        dictionary to update app.config with, useful for testing.
+    """
+    # The app
+    app = Flask(__name__)
 
-login_manager = LoginManager()
-login_manager.init_app(app)
-login_manager.login_view = 'login'
-
-csrf_protect.init_app(app)
+    # Load configdist for default values
+    app.config.from_object('configdist')
+    try:
+        # Attempt to update values with config.py
+        # API docs [0] show that from_object behaves like a dict update
+        # [0]: http://flask.pocoo.org/docs/0.11/api/#configuration
+        app.config.from_object('config')
+    except:
+        # Else keep the configdist defaults
+        pass
+
+    # Additional configuration options (used by testing)
+    from _15thnight.database import init_db
+
+    ### XXX: Override configuration options for testing here ###
+    app.config.update(cfg)
+
+    # XXX: must use app.db_session or current_app.db_session because of this
```

Comment:
Removed the XXX, though code documentation is important. Having too many comments is better than none. IMHO